lineman/app/components/pie_chart/pie_chart.coffee

```diff
@@ -0,0 +1,73 @@
+angular.module('loomioApp').directive 'pieChart', ->
+  template: '<div class="pie-chart"></div>'
+  replace: true
+  scope:
+    votes: '='
+  restrict: 'E'
+  controller: ($scope, $element) ->
+    size = $element.width()
+    paper = new Raphael($element[0], size, size)
+    cx = size / 2.0
+    cy = size / 2.0
+    r = (size / 2.0) - 2.0
+
+    console.log
+      size: size
+      cx: cx
+      cy: cy
+      r: r
+
+    # setting canvas scaling on element resize
+    paper.setViewBox(0, 0, size, size, true );
+    #paper.canvas.setAttribute('preserveAspectRatio', 'none');
+
+    paper.customAttributes.arc = (startAngle, endAngle) ->
+      rad = Math.PI / 180
+      x1 = cx + r * Math.cos(-startAngle * rad)
+      x2 = cx + r * Math.cos(-endAngle * rad)
+      y1 = cy + r * Math.sin(-startAngle * rad)
+      y2 = cy + r * Math.sin(-endAngle * rad)
+      path: ["M", cx, cy, "L", x1, y1, "A", r, r, 0, +(endAngle - startAngle > 180), 0, x2, y2, "z"]
+
+    positionColors =
+      yes: '#39A96F'
+      abstain: '#FAA030'
+      no: '#F15E72'
+      block: '#CE261B'
+
+    arcs = []
+
+    countUniquePositions = (votes) ->
+      _.sum votes, (num) -> +(num > 0)
+
+    $scope.$watch 'votes', ->
+      # reset start angle and remove existing shapes
+      start = 90
+      _.each arcs, (arc) -> arc.remove()
+
+      totalVotes = _.sum($scope.votes)
+      voteAngle = 360/totalVotes
+      totalSegments = countUniquePositions($scope.votes)
+      sortedPositions = _.sortBy(_.pairs($scope.votes), (pair) -> - pair[1])
+
+      switch totalSegments
+        when 0
+          arcs.push paper.circle(cx, cy, r).attr
+            'stroke-width': 0
+            fill: '#aaa'
+        when 1
+          position = sortedPositions[0][0]
+          arcs.push paper.circle(cx, cy, r).attr
+            'stroke-width': 0
+            fill: positionColors[position]
+        else
+          _.each sortedPositions, (pair) ->
```

Comment:
Hm, does this behave weirdly when there's only one vote? Seems the one vote case could work in the general case, unless it's leaving a little chunk of the pie out for some reason.